// @ngx-loading-bar/http/@ngx-loading-bar/http.es5.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Injectable, NgModule } from '@angular/core';
import { ConnectionBackend, Http, HttpModule, RequestOptions, XHRBackend } from '@angular/http';
import { LoadingBarModule, LoadingBarService } from '@ngx-loading-bar/core';
import { _finally as _finally$1 } from 'rxjs/operator/finally';
var LoadingBarHttp = /** @class */ (function (_super) {
    __extends(LoadingBarHttp, _super);
    /**
     * @param {?} _backend
     * @param {?} _defaultOptions
     * @param {?} loadingBar
     */
    function LoadingBarHttp(_backend, _defaultOptions, loadingBar) {
        var _this = _super.call(this, _backend, _defaultOptions) || this;
        _this.loadingBar = loadingBar;
        return _this;
    }
    /**
     * @param {?} url
     * @param {?=} options
     * @return {?}
     */
    LoadingBarHttp.prototype.request = function (url, options) {
        var _this = this;
        var /** @type {?} */ headers = options && options.headers;
        if (!headers && url && ((url)).headers) {
            headers = ((url)).headers;
        }
        if (headers && headers.has('ignoreLoadingBar')) {
            headers.delete('ignoreLoadingBar');
            return _super.prototype.request.call(this, url, options);
        }
        var /** @type {?} */ response$ = _super.prototype.request.call(this, url, options);
        if (options && options.ignoreLoadingBar === true) {
            return response$;
        }
        var /** @type {?} */ started = false;
        var /** @type {?} */ responseSubscribe = response$.subscribe.bind(response$);
        response$.subscribe = function () {
            var args = [];
            for (var _i = 0; _i < arguments.length; _i++) {
                args[_i] = arguments[_i];
            }
            _this.loadingBar.start();
            started = true;
            return responseSubscribe.apply(void 0, args);
        };
        return _finally$1.call(response$, function () { return started && _this.loadingBar.complete(); });
    };
    return LoadingBarHttp;
}(Http));
LoadingBarHttp.decorators = [
    { type: Injectable },
];
/**
 * @nocollapse
 */
LoadingBarHttp.ctorParameters = function () { return [
    { type: ConnectionBackend, },
    { type: RequestOptions, },
    { type: LoadingBarService, },
]; };
/**
 * @param {?} xhrBackend
 * @param {?} requestOptions
 * @param {?} loadingBarService
 * @return {?}
 */
function httpFactory(xhrBackend, requestOptions, loadingBarService) {
    return new LoadingBarHttp(xhrBackend, requestOptions, loadingBarService);
}
var LoadingBarHttpModule = /** @class */ (function () {
    function LoadingBarHttpModule() {
    }
    return LoadingBarHttpModule;
}());
LoadingBarHttpModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    HttpModule,
                    LoadingBarModule.forRoot(),
                ],
                exports: [
                    HttpModule,
                    LoadingBarModule,
                ],
                providers: [
                    { provide: Http, useFactory: httpFactory, deps: [XHRBackend, RequestOptions, LoadingBarService] },
                ],
            },] },
];
/**
 * @nocollapse
 */
LoadingBarHttpModule.ctorParameters = function () { return []; };
/**
 * Generated bundle index. Do not edit.
 */
export { LoadingBarHttpModule, LoadingBarHttp, httpFactory as ɵa };
